### ISSUE TITLE
Front End Editing Dock get stuck in edit mode

### ISSUE THREAD

**Issue by studiopatagon** [ExpressionEngine/ExpressionEngine#2769]
**Description of the problem**

Cannot disable 'edit' mode in Front End Editing Dock. Upon toggling it off, page gets reloaded with the toggle back on.

**How To Reproduce**

1. Turn on front end editing
2. Load side
3. Try to toggle edit mode off in the dock

**Screenshots / Videos / Template Code**
[Video of issue](https://willowlightstudio.com/assets/issue.mov)

**Environment Details:**
 - Version: 7.2.6
 - PHP Version: 7.2.36
 - MySQL Version: 5.6
 - OS: Linux
 - Web Server: Apache

**Possible Solution**
None known

**Comment by studiopatagon** [comment]
I've narrowed this down to a caching issue. If I'm in development mode and disable caching in the browser, then when the page reloads, it loads the value from exp_frontedit.

**Comment by intoeetive** [comment]
@willowlight to make sure, you're speaking about browser caching here, correct? Changing any of caching preferences in EE does not have effect here?

If this is correct, what do you do to enable / disable browser caching (and which browser do you use?)

**Comment by studiopatagon** [comment]
@intoeetive I open developer tools (both Safari and Chrome), go to the network tab and check "Disable cache". I've noticed that the exp_frontedit cookie has the correct value, but it looks like the dock does not read it always, unless caching is disabled.

**Comment by intoeetive** [comment]
do you have anything cache-related configured server side? Such as rules configured in `.htaccess` on nginx files, or all traffic going through caching server etc

**Comment by studiopatagon** [comment]
@intoeetive Vanilla .htaccess. I've tried this on several different sites and here's how I replicate it each time:

Load site, edit on the dock is on

1. Turn off editing, works
2. Turn on editing, works
3. Turn off editing, does not work. It's now stuck and only solution is to log out and log back in.

**Comment by jejuna** [comment]
I'm experiencing this same issue. In Firefox, Chrome AND Edge! The only way I can turn it off is to delete my browser cache. I have tried setting the lifetime settings for exp_frontedit cookie to 0 with no change. (I can't change that back to anything else now,)  I have also disabled caching in Caching area of Debugging and Output. No change. It would be nice to be able to set front-end ending on a per-channel basis.

**Comment by TomJaeger** [comment]
@jejuna What version of EE are you on?

**Comment by jejuna** [comment]
I'm on EE 7.213, PHP 8.1.9

**Comment by intoeetive** [comment]
@jejuna does the page reload after you click the toggle to turn front-end editing off (on front-end page)?
Do you see any javascript error in browser console?
Are you logged in as superadmin?

**Comment by jejuna** [comment]
@intoeetive Yes, the page does reload. Yes, I'm logged in a superadmin. I see no errors in the console.
